scipy/stats/tests/test_distributions.py

```diff
@@ -4414,6 +4457,39 @@ def test_ncx2_zero_nc_rvs():
     assert_allclose(result, expected, atol=1e-15)
 
 
+def test_ncx2_gh12731():
+    # test that gh-12731 is resolved; previously these were all 0.5
+    nc = 10**np.arange(5,10)
+    assert_allclose(stats.ncx2.cdf(1e4, df=1, nc=nc), 0)
```

Comment:
The true values are so small that the only possible correct answer when using 64 bit floating point is 0, so this should be `assert_equal`.
```suggestion
    assert_equal(stats.ncx2.cdf(1e4, df=1, nc=nc), 0)
```